core/src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -325,6 +325,9 @@ public static boolean parseMultiField(FieldMapper.Builder builder, String name,
 
             for (Map.Entry<String, Object> multiFieldEntry : multiFieldsPropNodes.entrySet()) {
                 String multiFieldName = multiFieldEntry.getKey();
+                if (multiFieldName.contains(".")) {
+                    throw new MapperParsingException("Field name [" + multiFieldName + "] cannot contain '.'");
```

Comment:
Maybe this message should give more context, that this is a multifield for another field (and include that field path)?